src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -0,0 +1,232 @@
+        String transactionId = pgsRequest.getIdTransaction();
+        try{
+            String authRequestJson = OBJECT_MAPPER.writeValueAsString(xPayRequest);
+            generateRequestEntity(clientId, mdcFields, transactionId, paymentRequestEntity, xPayRequest.getTimeStamp());
+            paymentRequestEntity.setJsonRequest(authRequestJson);
+            xPayRequest.setUrlRisposta(String.format(XPAY_RESPONSE_URL, paymentRequestEntity.getGuid()));
+        } catch (JsonProcessingException e) {
+            log.error(SERIALIZATION_ERROR_MSG, e);
+            return  createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+
+        try {
+            executeXPayAuthorizationCall(xPayRequest, paymentRequestEntity, transactionId);
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG, e);
+            return createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+        log.info("END - XPay Request Payment Authorization ");
+        return createxPayAuthResponse(null, HttpStatus.OK, paymentRequestEntity.getGuid());
+    }
+
+    @Async
+    private void executeXPayAuthorizationCall(AuthPaymentXPayRequest xPayRequest, PaymentRequestEntity paymentRequestEntity, String transactionId) {
+        log.info("START - execute XPay payment authorization call for transactionId: " + transactionId);
+        AuthPaymentXPayResponse response;
+        PaymentResponseEntity paymentResponseEntity = new PaymentResponseEntity();
+        try {
+            response = service.postForObject(xPayRequest);
+        } catch (ResourceAccessException e) {
+            log.warn("Timeout exception for xpay api request call");
+            throw e;
+        } catch (HttpServerErrorException hse){
+            log.warn("HttpServerErrorException exception for xpay api request call");
+            throw hse;
+        }
+
+        paymentRequestEntity.setXpayHtml(response.getHtml());
+        paymentRequestEntity.setTimeStamp(String.valueOf(response.getTimeStamp()));
+
+        generateResponseEntity(response, paymentResponseEntity, paymentRequestEntity.getGuid());
+        paymentResponseRepository.save(paymentResponseEntity);
+
+        paymentRequestRepository.save(paymentRequestEntity);
+        log.info("END - execute XPay payment authorization call for transactionId: " + transactionId);
+    }
+
+    private AuthPaymentXPayRequest createXPayRequest(XPayAuthRequest pgsRequest) {
+        log.info("create Request body to call XPay");
+        String codTrans = retrieveCodTrans(pgsRequest.getIdTransaction());
+        String timeStamp = String.valueOf(System.currentTimeMillis());
+        String mac = createMac(codTrans, pgsRequest.getGrandTotal(), timeStamp);
+        AuthPaymentXPayRequest xPayRequest = new AuthPaymentXPayRequest();
+        xPayRequest.setApiKey(API_KEY);
```

Comment:
@DavideAmatoNexi Since you are reading this value from the environment, maybe its value should be evaluated before setting it in the request body.